presto-main-base/src/main/java/com/facebook/presto/server/thrift/ConnectorSplitThriftCodec.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.server.thrift;
+
+import com.facebook.airlift.json.JsonCodec;
+import com.facebook.drift.codec.CodecThriftType;
+import com.facebook.drift.codec.metadata.ThriftType;
+import com.facebook.drift.protocol.TProtocolReader;
+import com.facebook.drift.protocol.TProtocolWriter;
+import com.facebook.presto.connector.ConnectorThriftCodecManager;
+import com.facebook.presto.metadata.HandleResolver;
+import com.facebook.presto.spi.ConnectorSplit;
+
+import javax.inject.Inject;
+
+import java.nio.ByteBuffer;
+
+import static java.util.Objects.requireNonNull;
+
+public class ConnectorSplitThriftCodec
+        extends AbstractTypedThriftCodec<ConnectorSplit>
+{
+    private static final ThriftType THRIFT_TYPE = createThriftType(ConnectorSplit.class);
+    private final ConnectorThriftCodecManager connectorThriftCodecManager;
+
+    @Inject
+    public ConnectorSplitThriftCodec(HandleResolver handleResolver, ConnectorThriftCodecManager connectorThriftCodecManager, JsonCodec<ConnectorSplit> jsonCodec)
+    {
+        super(ConnectorSplit.class,
+                requireNonNull(jsonCodec, "jsonCodec is null"),
+                requireNonNull(handleResolver, "handleResolver is null")::getId,
+                handleResolver::getSplitClass);
+        this.connectorThriftCodecManager = requireNonNull(connectorThriftCodecManager, "connectorThriftCodecManager is null");
+    }
+
+    @CodecThriftType
+    public static ThriftType getThriftType()
+    {
+        return THRIFT_TYPE;
+    }
+
+    @Override
+    public ThriftType getType()
+    {
+        return THRIFT_TYPE;
+    }
+
+    @Override
+    public ConnectorSplit readConcreteValue(String connectorId, TProtocolReader reader)
+            throws Exception
+    {
+        byte[] bytes = reader.readBinary().array();
```

Comment:
`.array()` may not always work. `ByteBuffer` may have an offset and a length set to be other than `0` and `array.length`. You need to consider this or at least add an explicit assert. 

Unfortunately with the current API of `codec.deserialize(bytes)` you may need to copy if the offset is not `0` or the length is not `array().length`. A better API could be something like `byte[] array, int offset, int length`, but we keep keep it as is for now as we are planning on changing it anyways to optimize allocations.